### PR TITLE
Post api articles article id comments

### PR DESCRIPTION
**Pull request by JohnDMurphy**
-Post new data to comments
-Handle errors

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -218,5 +218,37 @@ describe('nc-be-news-app', () => {
       // Check the amount of items in the object is correct
       expect(Object.keys(comment).length).toBe(6);
```
Comment (cavalierswain) [inline __tests__/app.test.js:218]: You can also check to see what you are getting back in the comment body.

File: controllers/getArticleById.controllers.js [hunk 2]
```diff
@@ -65,9 +65,13 @@ exports.postCommentById = async (req, res, next) => {
     const { article_id } = req.params;
     const { author, body } = req.body;
 
-    const postedData = await addNewComment(article_id, author, body);
+    if (typeof author !== 'string' || typeof body !== 'string') {
+      next({ status: 400, msg: 'Incorrect Input Type' });
+    } else {
```
Comment (cavalierswain) [inline controllers/getArticleById.controllers.js]: ideally we would want to keep the error handling out of the controller, maybe consider moving this into the model

Comment (JohnDMurphy) [reply controllers/getArticleById.controllers.js]: done

File: __tests__/app.test.js (hunk 1 above)
Comment (JohnDMurphy) [reply __tests__/app.test.js:218]: done

**Comment by cavalierswain** [comment]
Good refactoring for this, well done on implementing the changes!

**Pull request merged**